### ISSUE TITLE
IRF interpolation outside node ranges

### ISSUE THREAD

**Issue by juanjq** [cta-observatory/magic-cta-pipe#275]
I noticed in the limiting cases, when for example, in high ZD observations. If your target cosZD is outside of the range that your grid points cover, even that if the distance is relatively small. Then the aeff interpolation is failing (i.e. returning all nan), and consequently when trying to create the hdu (line 373 of /magic-cta-pipe/magicctapipe/scripts/lst1_magic/lst1_magic_dl2_to_dl3.py) program fails with the following error "IndexError: index 0 is out of bounds for axis 0 with size 0" and without any other explanation.

Can be solved providing only one IRF in those cases, but I think something can be added/changed in the code. Selecting just closer node in this cases, and no interpolation, or at least tell the user that the hdu creation failed because the observations cannot be interpolated in this case.

If you need it I can prepare a small script with some data with these conditions for you to recreate the error.

**Comment by Elisa-Visentin** [comment]
I can have a look into the dl2-dl3 script to check what is happening (not now). Just one comment: maybe we should use the nearest node only if it is within XXX deg from the target (otherwise, don't produce the output file), just to prevent cases in which, due to missing test nodes, you assign to your observation a IRF evaluated too far away (e.g., observation at 50 deg, IRFs only up to 20 deg, you use the 20 deg IRF for a 50 deg observation: quite dangerous!)

**Comment by juanjq** [comment]
I agree, even that is close I guess make sense at least notify that no interpolation is being used, and in the case that is too far, notify that output is not being produced because of that. About the limiting distance maybe makes sense to use the config parameter 'dl2_to_dl3': {'max_distance': }? Or maybe is too large for this purpose, this I don't know...
